src/bsoncxx/string/view_or_value.hpp

```diff
@@ -0,0 +1,77 @@
+// Copyright 2015 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <bsoncxx/config/prelude.hpp>
+
+#include <string>
+
+#include <bsoncxx/stdx/string_view.hpp>
+#include <bsoncxx/view_or_value.hpp>
+
+#include <iostream>
+
+namespace bsoncxx {
+BSONCXX_INLINE_NAMESPACE_BEGIN
+namespace string {
+
+using string_variant = bsoncxx::view_or_value<stdx::string_view, std::string>;
```

Comment:
Should anyone be using the string_variant type directly? If not, maybe don't make the public typedef. It will make the declarations below a little worse, but you can always make a private typedef in view_or_value below so you can at least refer to the super class constructors easily.